### DISCUSSION TITLE
Has anyone implemented a drop animation without using DragOverlay?

### DISCUSSION THREAD

**Discussion by jmcannon** [clauderic/dnd-kit#679] [Q&A]
I feel like I'm almost there, but still getting jumpy behavior when I release.

I'm setting a conditional style like this on my SortableItem:

`transition: isDragging ? '' : 'transform 350ms'`

Any nudges in the right direction are appreciated! Thanks :)

**Comment by joseDaKing** [comment]
You could hide the actual Component while dragging and use drag overlay it will yield great results

**Comment by clauderic** [comment]
@jmcannon I'd recommend using `<DragOverlay>` if you plan on having drop animations. They're really hard to achieve without it. What's the rationale for not using `<DragOverlay>`?

**Reply by jmcannon** [reply]
Thanks for the response! I have used DragOverlay for other things, and it works well. My question pertains to a feature where I have draggable videos. I don't want to load two HTML video elements per video. I could make a static image of the first frame of each video, but that's a lot of work and doesn't account for dragging a video that's not at the beginning.

I'm open to other ideas for a clean DragOverlay implementation with videos, but the Sortable works great for this use case out of the box. It would be really nice to have that clean drop animation, but it's not a deal-breaker.

**Reply by RemyMachado** [reply]
Any update on this? This [story example](https://master--5fc05e08a4a65d0021ae0bf2.chromatic.com/?path=/story/core-droppable-usedroppable--basic-setup) doesn't use an overlay and has a nice animation. Is it still hard to achieve?

EDIT: it turns out that it uses `DragOverlay` under the hood, without mentioning it in the story title...

**Reply by timurridjanovic** [reply]
Any solution for this? @clauderic @jmcannon @RemyMachado 

I can't use the DragOverlay because I'm dealing with black box components (I can't change them) that reload everytime they're unmounted/remounted. If I render them in the DragOverlay, it looks weird because the component Im dragging suddenly reloads and shows spinners. Not using the DragOverlay works but then the drop animation looks choppy..
